docs/user/alerting/alerting-troubleshooting.asciidoc

```diff
@@ -2,234 +2,205 @@
+The result of this http request (and printed to stdout by https://github.com/pmuellr/kbn-action[kbn-action]) will be data returned by the action execution, along with error messages if errors were encountered.
 
 [float]
-[[connector-tls-settings]]
-=== Connectors have TLS errors when executing actions
-
-*Problem*:
+[[alerting-framework-health-ui]]
+=== Alerting Framework Health
 
-When executing actions, a connector gets a TLS socket error when connecting to
-the server.
+Rule Management and Rule Details pages contains an error banner, which helps to identify the errors for the rules:
+[role="screenshot"]
+image::images/rules-management-health.png[Rule management page with the errors banner]
 
-*Resolution*:
-
-Configuration options are available to specialize connections to TLS servers,
-including ignoring server certificate validation, and providing certificate
-authority data to verify servers using custom certificates.  For more details, 
-see <<action-settings,Action settings>>.
+[role="screenshot"]
+image::images/rules-details-health.png[Rule details page with the errors banner]
 
 [float]
-[[rules-long-execution-time]]
-=== Identify long-running rules
+[[task-manager-diagnostics]]
+=== Task Manager diagnostics
 
-The following query can help you identify rules that are taking a long time to execute and might impact the overall health of your deployment.
+Under the hood, Rules and Connectors use another plugin called Task Manager, which handles the scheduling, execution and error handling of the tasks that drive the Rules and Connectors features.
+This means that failure cases in Rules or Connectors will, at times, be revealed by the Task Manager mechanism, rather than the Rules mechanism.
 
-[IMPORTANT]
-==============================================
-By default, only users with a `superuser` role can query the {kib} event log because it is a system index. To enable additional users to execute this query, assign `read` privileges to the `.kibana-event-log*` index.
-==============================================
+Task Manager provides a visible status which can be used to diagnose issues and is very well documented health monitoring and troubleshooting. 
+Task Manager uses the `.kibana_task_manager index`, which is an internal index we use to contain all the saved objects that represent the tasks in the system.
 
-Query for a list of rule ids, bucketed by their execution times:
+[float]
+==== Getting from a Rule, to its Task
+When a Rule is created a Task is created which is scheduled to run at the interval specified. This means that if, for example, when a rule is created its configured to check every 5 minutes, then the underlying will be expected to run every 5 minutes (in practice, after each time the alert is ran, we’ll schedule the task to run again in 5 minutes, rather than it being scheduled to run itself every 5 minutes indefinitely).
```

Comment:
```suggestion
When a rule is created, a task is created, scheduled to run at the interval specified. For example, when a rule is created and configured to check every 5 minutes, then the underlying task will be expected to run every 5 minutes. In practice, after each time the alert is run, the task is scheduled to run again in 5 minutes, rather than being scheduled to run every 5 minutes indefinitely.
```